.circleci/config.yml

```diff
@@ -329,6 +333,115 @@ jobs:
             - todo-api
             - todo-frontend
 
+  aws-generate-terraform-language-files:
+    executor: ubuntu-machine
+    steps:
+      - checkout-opta
+      - attach_workspace:
+          at: ./
+      - install-opta-dependencies
+      - run:
+          name: "Generate Terraform Language Files"
+          command: |
+            mkdir $HOME/generated-tf-files-aws
+            ./dist/opta/opta generate-terraform \
+            --config ./.circleci/ci-tests/terraform-generated-aws/environment.yaml \
+            --directory $HOME/generated-tf-files-aws \
+            --auto-approve
+            ./dist/opta/opta generate-terraform \
+            --config ./.circleci/ci-tests/terraform-generated-aws/service-http.yaml \
+            --directory $HOME/generated-tf-files-aws \
+            --auto-approve
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - generated-tf-files-aws
+
+  aws-tg-create-env:
+    executor: ubuntu-machine
+    steps:
+      - install-opta-dependencies
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: "Create Terraform Generated Environment"
+          command: |
+            cd $HOME/generated-tf-files-aws
+            terraform init
+            terraform plan -compact-warnings -lock=false -input=false -out=tf.plan -target=module.base
+            terraform apply -compact-warnings -auto-approve tf.plan
+            cat ./tfstate/awsenv-tg.tfstate
```

Comment:
why do we need the cat here?
it will make the log too large

Reply:
This is a temporary solution so that if there is a fail, we would have the state files if need arises to pull the state in a different machine, since currently it would be stored locally, that that would just go away since the CI machines are ephemeral.

Reply:
you mean to troubleshoot? 
we can use artifacts for that https://circleci.com/docs/2.0/artifacts/